pkg/jaeger/query/trace_query.go

```diff
@@ -108,42 +108,63 @@ const (
 	`
 )
 
-func buildCompleteTraceQuery(traceIDClause string) string {
+type Builder struct {
+	cfg *Config
+}
+
+func NewBuilder(cfg *Config) *Builder {
+	return &Builder{cfg}
+}
+
+func (b *Builder) buildCompleteTraceQuery(traceIDClause string, timeLowRef string, timeHighRef string) string {
+	spanTimeClause := ""
+	eventTimeClause := ""
+	linkTimeClause := ""
+	if timeLowRef != "" {
```

Comment:
A nit: the way `timeLowRef` and `timeHightRef` are used I'd rather prefer a single boolean flag in the `buildCompleteTraceQuery` signature.

Reply:
If there was a boolean then buildCompleteTraceQuery would need to know the names " "trace_ids.time_low", "trace_ids.time_high" but I don't think that function should know anything about the trace_ids table. That's why I think passing in the names is cleaner

Reply:
Good point! I realized I can add clauses independently for timeLowRef and timeHighRef. This allows the api for one to be set but not the other and prevents only one from being used at the sentinel. Makes the code more robust. Thanks!